File: robocode-tankroyale-bot-api-java/src/main/java/dev/robocode/tankroyale/botapi/events/RoundEndedEvent.java
package dev.robocode.tankroyale.botapi.events;

/** Event occurring when a round has just ended. */
@SuppressWarnings("unused")
public final class RoundEndedEvent extends BotEvent {

  /** The round number. */
  private final int roundNumber;

  /**
   * Initializes a new instance of the RoundEndedEvent class.
   *
   * @param roundNumber is the round number.
   * @param turnNumber is the turn number.
   */
  public RoundEndedEvent(int roundNumber, int turnNumber) {
    super(turnNumber);
    this.roundNumber = roundNumber;
  }

  /**
   * Returns the round number.
   *
   * @return The round number.
   */
  public int getRoundNumber() {
    return roundNumber;
  }
}

File: robocode-tankroyale-bot-api-dotnet/src/events/RoundEndedEvent.cs
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi.Events
{
  /// <summary>
  /// Event occurring when a round has just ended.
  /// </summary>
  public sealed class RoundEndedEvent : IEvent
  {
    /// <summary>The round number.</summary>
    public int RoundNumber { get; }

    /// <summary>The turn number.</summary>
    public int TurnNumber { get; }

    /// <summary>
    /// Initializes a new instance of the RoundEndedEvent class.
    /// </summary>
    /// <param name="roundNumber">The round number</param>
    /// <param name="turnNumber">The turn number.</param>
    [JsonConstructor]
    public RoundEndedEvent(int roundNumber, int turnNumber) : base() =>
      (RoundNumber, TurnNumber) = (roundNumber, turnNumber);
  }
}